config/routes.rb

```diff
@@ -100,6 +100,14 @@
           end
         end
 
+        resources :vcs_clone, :only => [] do
+          collection do
+            get 'get_repo_information', to: 'vcs_clone#repo_information'
+            post 'install_role', to: 'vcs_clone#install_role'
+            get 'get_installed_roles', to: 'vcs_clone#installed_roles'
```

Comment:
REST encodes methods in the HTTP method, not in the path. So I'd think this is a more natural API:
```suggestion
            get 'repo_information', to: 'vcs_clone#repo_information'
            post 'roles', to: 'vcs_clone#install_role'
            get 'roles', to: 'vcs_clone#installed_roles'
```

You `POST` to `roles` to create a new one and `GET` to `/roles` to get the installed roles.

Then uninstall would be `DELETE /roles/:id` and updating `PUT /roles/:id`.

Reply:
Indeed, this makes more sense semantically.
I moved the functionality behind a single endpoint with GET, POST, PUT and DELETE methods.